packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -280,7 +280,8 @@ class AttachCommand extends FlutterCommand {
             devicePort: deviceVmservicePort,
             hostPort: hostVmservicePort,
           );
-        globals.printStatus('Waiting for a connection from Flutter on ${device.name}...');
+        globals.printStatus('If your app is already running, please restart it (or pass --debug-uri <observatory_url>)');
```

Comment:
I don't think we want this message showing up for platforms where they don't need to restart, or a restart wouldn't help because we can't back-read logs.